### PR TITLE
fix ms windows compatibility

### PR DESCRIPTION
**Pull request by iTytar**
modified:   spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/service/PackageMetadataService.java
	modified:   spring-cloud-skipper-server-core/src/main/resources/application.yml

### CODE REVIEW

File: spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/service/PackageMetadataService.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ private boolean checkIfPackageIsFromLocalRepo(List<String> errorMessages, Packag
 				try {
 					if (!packageRepository.isLocal()) {
 						Resource resource = resourceLoader.getResource(packageRepository.getUrl()
-								+ File.separator + "index.yml");
+								+ "/index.yml");
```
Comment (ilayaperumalg) [inline spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/service/PackageMetadataService.java]: Isn't the `file.separator` working in your case? Could you elaborate on your changes?

File: spring-cloud-skipper-server-core/src/main/resources/application.yml [hunk 2]
```diff
@@ -36,7 +36,12 @@ spring:
               checksum-sha1-url: "{repository}/org/springframework/cloud/spring-cloud-skipper-shell/{version}/spring-cloud-skipper-shell-{version}.jar.sha1"
               checksum-sha256-url: "{repository}/org/springframework/cloud/spring-cloud-skipper-shell/{version}/spring-cloud-skipper-shell-{version}.jar.sha256"
         package-repositories:
-          local:
+          -
```
Comment (ilayaperumalg) [inline spring-cloud-skipper-server-core/src/main/resources/application.yml]: We removed this on [purpose](https://github.com/spring-cloud/spring-cloud-skipper/commit/d7f9baf615bc801b7e4cb38e77421653d0848cc0#diff-d23f8c4327c765ff9329c887fb12b5ec): https://github.com/spring-cloud/spring-cloud-skipper/issues/762

**Comment by ilayaperumalg** [comment]
@iTytar Thanks for your contribution!

File: spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/service/PackageMetadataService.java (hunk 1 above)
Comment (iTytar) [reply spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/service/PackageMetadataService.java]: Do not use a local (OS specific) value of File.separator when you build URL to a remote service.

Comment (iTytar) [reply spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/service/PackageMetadataService.java]: File.separator isn't URL separator in common case. For example, windows File.separator is '\'. Result in this case will be <repository URL>\index.yml. This isn't valid URL. And you get error.

Comment (ilayaperumalg) [reply spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/service/PackageMetadataService.java]: @iTytar Now I see the issue. Thanks for the clarification. Will merge this PR after excluding the application.yml change.

**Comment by ilayaperumalg** [comment]
Rebased, amended and merged the PR as 1155150.

@iTytar Thank you again for your contribution!

**Pull request closed without merging**